src/primitives/arc/mod.rs

```diff
@@ -106,9 +107,51 @@ impl Arc {
         Circle::new(self.top_left, self.diameter)
     }
 
-    /// Return the center point of the arc.
+    /// Returns the center point of the arc.
     pub fn center(&self) -> Point {
-        self.bounding_box().center()
+        self.to_circle().center()
+    }
+
+    fn radius(&self) -> Real {
+        Real::from(self.diameter.saturating_sub(1)) / 2.into()
+    }
+
+    /// Returns the end angle of the arc.
+    pub fn angle_end(&self) -> Angle {
+        self.angle_start + self.angle_sweep
+    }
+
+    /// Returns the Point from a certain angle.
+    pub fn point_from_angle(&self, angle: &Angle) -> Point {
+        let center = self.center();
+        let radius = self.radius();
```

Comment:
To get accurate results you'll need to use `Circle::center_2x` instead of `center` and scale all coordinates by 2. `center` does round the center point to the nearest integer, which isn't accurate for circles/arcs with even diameters. The radius in the scaled up coordinate system is then equal to `diameter`.

Reply:
I feel like dividing the returned `Point` by 2 will have the same effect, isn't it?

Reply:
Which `Point` do you mean? The result of `point_from_angle`?

Reply:
Yes. I feel I could avoid to work with the radius on the `bounding_box` function and simply reuse `top_left + diameter`, etc.

Reply:
I'm still not sure I understand what you mean. But you could get rid of using `Real` for the `radius` function by scaling the coordinate system up by a factor of 2.

Reply:
```rust
let arc = Arc::new(Point::new(10, 10), 5, 0.0.deg(), 90.0.deg());
let center = self.to_circle().center_2x();
```

center is `Point { x: 24, y: 24 }`, and it should be `Point { x: 25, y: 25 }`

Reply:
`(24, 24)` or `(12, 12)` (unscaled) is the correct center. Here is a diagram that shows a circle with top left at `(10, 10)` and a diameter of 5:

![grafik](https://user-images.githubusercontent.com/226123/196282670-8fd9e5c4-53e4-4f09-a08d-f688a673878e.png)

Integer coordinates in e-g are centered inside the corresponding pixel (blue dots in the diagram). But the outside of filled shapes are defined to be aligned with the boundaries between pixels (grey grid lines in the diagram).

Reply:
Thanks a lot for the help and explanation. Not easy 😅
I pushed the changes.

Reply:
> Not easy sweat_smile

No it's not, the half pixel offset that creeps into many calculations can get really confusing.

The BB isn't correct for some negative start angles:
![grafik](https://user-images.githubusercontent.com/226123/196668204-cdd48356-d40f-4097-a798-551c26fb1695.png)

Reply:
Negative sweep angles are also broken:
![grafik](https://user-images.githubusercontent.com/226123/196669099-5cf86290-a7d9-40c7-9f16-854ee82fa3eb.png)

Reply:
Oh, completely forgot to test the negatives angles 😅
But I've seen that I forgot to normalize the angles during the refactoring, which is probably why it's broken. (hopefully)

Reply:
Ok, I fixed it by using the PlaneSector that @jamwaffles introduced on the other PR. Although it's not always pixel perfect.

With the debug tools, the stroke size pushes the bounding box in a weird way. Shouldn't it stay the same, disregarding the stroke width?